aspnetcore/web-api/index.md

```diff
@@ -1,22 +1,135 @@
 ---
-title: Build Web APIs in ASP.NET Core
+title: Build web APIs with ASP.NET Core
 author: scottaddie
-description: A collection of resources related to using ASP.NET Core Web API
+description: Learn about the features available for building a web API in ASP.NET Core and when it's appropriate to use each feature.
 manager: wpickett
 ms.author: scaddie
-ms.date: 04/02/2018
-ms.prod: asp.net-core
+ms.custom: mvc
+ms.date: 04/23/2018
+ms.prod: aspnet-core
 ms.technology: aspnet
 ms.topic: article
 uid: web-api/index
 ---
-# Build Web APIs in ASP.NET Core
+# Build web APIs with ASP.NET Core
 
-* [Create a Web API with ASP.NET Core and Visual Studio Code](xref:tutorials/web-api-vsc)
-* [Create a Web API with ASP.NET Core and Visual Studio for Mac](xref:tutorials/first-web-api-mac)
-* [Create a Web API with ASP.NET Core and Visual Studio for Windows](xref:tutorials/first-web-api)
-* [Controller action return types in ASP.NET Core Web API](xref:web-api/action-return-types)
-* [Help pages using Swagger](xref:tutorials/web-api-help-pages-using-swagger)
-* [Create backend services for native mobile apps](xref:mobile/native-mobile-backend)
+By [Scott Addie](https://github.com/scottaddie)
+
+[View or download sample code](https://github.com/aspnet/Docs/tree/master/aspnetcore/web-api/define-controller/samples) ([how to download](xref:tutorials/index#how-to-download-a-sample))
+
+ASP.NET Core offers the following options for creating a web API controller:
+
+::: moniker range="<= aspnetcore-2.0"
+* [Derive class from Controller](#derive-class-from-controller)
+* [Derive class from ControllerBase](#derive-class-from-controllerbase)
+::: moniker-end
+::: moniker range=">= aspnetcore-2.1"
+* [Derive class from Controller](#derive-class-from-controller)
+* [Derive class from ControllerBase](#derive-class-from-controllerbase)
+* [Annotate class with ApiControllerAttribute](#annotate-class-with-apicontrollerattribute)
+::: moniker-end
+
+This document explains when it's most appropriate to use each option.
+
+## Derive class from Controller
+
+Inherit from the [Controller](/dotnet/api/microsoft.aspnetcore.mvc.controller) class in a controller that needs to support HTML and Razor in addition to web API actions. Examples that require `Controller` inheritance include returning MVC views or [invoking view components](xref:mvc/views/view-components#invoking-a-view-component-directly-from-a-controller).
```

Comment:
>in a controller that needs to support HTML and Razor in addition to web API actions. 

I don't really know what that's a topic. You certainly could write a controller that serves both APIs and views, but I don't think we'd recommend that users do this on purpose.

Reply:
Regardless of what it used to derive from, I don't understand why we would discuss it as `a controller that needs to support HTML and Razor in addition to web API actions` - this isn't something we've ever really told people to do. It's explicitly recommended or disrecommended, which is why I'm suprised to see it getting coverage here.

Reply:
@danroth27 Thoughts?

Reply:
:+1: to @rynowak's comment. I don't think we need to discuss deriving from `Controller` in this topic or anything views related.